rendermodules/stack/swap_zs.py

```diff
@@ -34,110 +34,51 @@ def swap_section(render, source_stack, target_stack, z, pool_size=5):
     source_ts = renderapi.resolvedtiles.get_resolved_tiles_from_z(source_stack, z, render=render)
     target_ts = renderapi.resolvedtiles.get_resolved_tiles_from_z(target_stack, z, render=render)
 
-    temp_stacks = []
-    # write the tilespecs to a temp stack
-    temp_stack1 = "em_2d_montage_solved_swap_temp_stack_{}_z{}_t{}".format(source_stack, z, time.strftime("%m%d%y_%H%M%S"))
-    temp_stack2 = "em_2d_montage_solved_swap_temp_stack_{}_z{}_t{}".format(target_stack, z, time.strftime("%m%d%y_%H%M%S"))
+    source_tileids = [ts.tileId for ts in source_ts.tilespecs]
+    target_tileids = [ts.tileId for ts in target_ts.tilespecs]
 
-    try:
-        render.run(renderapi.stack.create_stack,
-                   temp_stack1)
-        render.run(renderapi.stack.create_stack,
-                   temp_stack2)
-    except Exception as e:
-        print("Cannot create temp stacks")
+    if set(source_tileids) == set(target_tileids):
```

Comment:
This might even need a stronger assertion, since any duplicate tileids will be lost in both stacks
```suggestion
    if set(source_tileids) & set(target_tileids):
```